etc/profile-a-l/build-systems-common.profile

```diff
@@ -0,0 +1,65 @@
+# Firejail profile for build-systems-common
+# This file is overwritten after every install/update
+# Persistent local customizations
+include build-systems-common.local
+# Persistent global definitions
+# added by caller profile
+#include globals.local
+
+ignore noexec ${HOME}
+ignore noexec /tmp
+
+# Allow /bin/sh (blacklisted by disable-shell.inc)
+include allow-bin-sh.inc
+
+# Allows files commonly used by IDEs
+include allow-common-devel.inc
+
+# Allow ssh (blacklisted by disable-common.inc)
+#include allow-ssh.inc
+
+blacklist ${RUNUSER}
+
+include disable-common.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-programs.inc
+include disable-shell.inc
+include disable-X11.inc
+include disable-xdg.inc
+
+whitelist ${HOME}/Projects
```

Comment:
Considering that there is no standard path for this, if the includers of
build-systems-common.profile are ever enabled by default, I'd wager that there
would be many issues opened due to the whitelist above.  I suppose that we
could at least add comments like:

```
# Note: Be sure to whitelist your own development project directories:
#whitelist ${HOME}/YOUR_PROJECT_DIR
```

This is kind of similar to what was added to npm.profile on #3866:

>     # If you want whitelisting, change the line below to your npm projects directory
>     # and uncomment the lines below.
>     [...]
>     #whitelist ${HOME}/Projects

Which was later changed to the following on #3876:

>     # If you want whitelisting, change ${HOME}/Projects below to your npm projects directory
>     # and uncomment the lines below.
>     [...]
>     #whitelist ${HOME}/Projects

(And which now lives on nodejs-common.profile since #4255)

Cc: @aidalgol @glitsj16

Reply:
IMHO we must not ever sandbox build-systems by default as they can require anything.

I don't even sandbox IDEs. The profile are very weak / must be weak (full D-Bus, no whitelisting, minimal blacklisting, ...) the win is very low to nothing but the sandbox regularly breaks things will developing and since nesting isn't supported you can not sandbox the things which really need sandboxing in a tight profile.

Reply:
I am in favour of changing this to make the whitelisting opt-in just as in `nodejs-common.profile`, and I think `nodejs-common.profile` should be refactored as part of this set of changes in the same way the other package manager profiles have been.

Reply:
My idea was that since you already need to type `firejail meson compile -C _builddir` (for example), using `firejail --whitelist=~/my_project_dir meson compile -C _builddir` isn't much more complicated. Anyway, will make whitelisting opt-in.

Reply:
@rusty-snake [on Sep 10](https://github.com/netblue30/firejail/pull/4519#discussion_r706252880):

> Anyway, will make whitelisting opt-in.

Thanks.  Can you still put a comment above it noting that it's intended to be
changed?

It's not clear to me that it's supposed to be user-customized; I think that the
path could be easily mistaken as something related to a build system.